Segment-Intercom/Classes/SEGIntercomIntegration.m

```diff
@@ -74,10 +74,48 @@ - (void)track:(SEGTrackPayload *)payload
         SEGLog(@"[Intercom logEventWithName:%@];", payload.event);
         return;
     }
-    [self.intercom logEventWithName:payload.event metaData:payload.properties];
-    SEGLog(@"[Intercom logEventWithName:%@ metaData:%@];", payload.event, payload.properties);
+
+
+    NSMutableDictionary *original = [NSMutableDictionary dictionaryWithDictionary:payload.properties];
+    NSMutableDictionary *output = [NSMutableDictionary dictionaryWithCapacity:payload.properties.count];
+
+    NSMutableDictionary *price = [NSMutableDictionary dictionaryWithCapacity:0];
+
+    [original enumerateKeysAndObjectsUsingBlock:^(id key, id data, BOOL *stop) {
+
+        [output setObject:data forKey:key];
+        if ([key isEqual:@"revenue"] || [key isEqual:@"total"]) {
+            double dataValue = [data doubleValue];
+            long amountInCents = dataValue * 100;
+            NSNumber *finalAmount = [[NSNumber alloc] initWithLong:amountInCents];
+            [price setObject:finalAmount forKey:@"amount"];
+
+            [original removeObjectForKey:@"revenue"];
```

Comment:
Why is it necessary to remove these values from original? Can't you just remove them from the output object?

Reply:
Because it will continue to iterate on the object and see that `total` is present, overriding the `revenue` value in `amount` for `total`